Homebrew 2.6.0 (#559)
Homebrew/brew.sh: _posts/2020-11-31-homebrew-2.6.0.md

---
title: 2.6.0
author: MikeMcQuaid
---

> Today I'd like to announce Homebrew 2.6.0. The most significant changes since 2.5.0 are macOS Big Sur support, `brew` commands replacing all `brew cask` commands, the beginnings of macOS M1/Apple Silicon/ARM support and API deprecations.

Reviewer: `macOS Big Sur support on Intel`

I know that you talk about M1 at the end of the sentence, but as this is the first thing people will read, I would not like to raise wrong expectations.

> Today I'd like to announce Homebrew 2.6.0. The most significant changes since 2.5.0 are macOS Big Sur support, `brew` commands replacing all `brew cask` commands, the beginnings of macOS M1/Apple Silicon/ARM support and API deprecations.

Author: ```suggestion
Today I'd like to announce Homebrew 2.6.0. The most significant changes since 2.5.0 are macOS Big Sur support on Intel, `brew` commands replacing all `brew cask` commands, the beginnings of macOS M1/Apple Silicon/ARM support and API deprecations.
```

Major changes and deprecations since 2.5.0:

- [macOS Big Sur is supported (and High Sierra unsupported)](https://github.com/Homebrew/brew/pull/9054)
- [`depends_on :java`, `brew switch`, `brew diy` and various other APIs have been deprecated](https://github.com/Homebrew/brew/pull/9209)

> - [All `brew cask` commands have been deprecated in favour of `brew` commands (with `--cask`) when necessary](https://github.com/Homebrew/brew/pull/8899)

Reviewer 2: I've noticed a few people reporting issues related to a change in `brew list` functionality that was deprecated in homebrew/brew#8851 (released in 2.5.7) and I believe it's intended to match the deprecation of `brew cask` commands. However, I believe the `odeprecated` calls from homebrew/brew#8851 were included in the conversions to `odisabled` in homebrew/brew#9209. I'm going to open a PR to narrowly un-`odisable` the changes from homebrew/brew#8851

Reviewer 2: https://github.com/Homebrew/brew/pull/9357

Author: Merged!

- [`Homebrew.args` is deprecated](https://github.com/Homebrew/brew/pull/8795)
- [All `Requirement`s are deprecated in Homebrew/core](https://github.com/Homebrew/brew/pull/9052)

> - [macOS Homebrew on M1/Apple Silicon/ARM has partial functionality. We recommend installing into `/opt/homebrew` and forbid installing into `/usr/local` (to avoid clashing with the macOS Intel install and allow their usage side-by-side).](https://github.com/Homebrew/brew/pull/9117)

Reviewer 3: ```suggestion
- [macOS Homebrew on M1/Apple Silicon/ARM has partial functionality. We recommend installing into `/opt/homebrew` and forbid installing into `/usr/local` (to avoid clashing with the macOS Intel install and allow their usage side-by-side)](https://github.com/Homebrew/brew/pull/9117).
```

Reviewer 4: ```suggestion
- [macOS Homebrew running natively on M1/Apple Silicon/ARM has partial functionality. We recommend installing into `/opt/homebrew` and forbid installing into `/usr/local` (to avoid clashing with the macOS Intel install and allow their usage side-by-side)](https://github.com/Homebrew/brew/pull/9117). We currently recommend running Homebrew using Intel emulation with Rosetta 2.
```

> - [`brew tap-new` will setup GitHub Actions workflows to](https://github.com/Homebrew/brew/pull/8762) [upload to GitHub Releases](https://github.com/Homebrew/brew/pull/8410). Read [the blog post](https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/) for more documentation.

Reviewer 5: ```suggestion
- [`brew tap-new` will set up GitHub Actions workflows to](https://github.com/Homebrew/brew/pull/8762) [upload to GitHub Releases](https://github.com/Homebrew/brew/pull/8410). Read [the blog post](https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/) for more documentation.
```

- [GitHub deprecated their API's basic authentication](https://github.com/Homebrew/brew/pull/8826)

Other changes since 2.5.0 I'd like to highlight are the following:

> - [`brew doctor` checks all tap branches, not just `homebrew-core`](https://github.com/Homebrew/brew/pull/9305)

Reviewer 6: Is homebrew-core a branch in this?

Reviewer 7: The wording's slightly hard to parse without looking it up - I had to look at the PR to interpret what this one meant. How about this wording?
```suggestion
- [`brew doctor` checks the active branch for all taps, not just `homebrew-core`](https://github.com/Homebrew/brew/pull/9305)
```

> - [`brew doctor` checks all tap branches, not just `homebrew-core`](https://github.com/Homebrew/brew/pull/9305)

Author: The wording's slightly hard to parse without looking it up - I had to look at the PR to interpret what this one meant. How about this wording?
```suggestion
- [`brew doctor` checks the active branch for all taps, not just Homebrew/homebrew-core](https://github.com/Homebrew/brew/pull/9305)
```

- [`brew unbottled` is a new developer command to identify formulae that haven't had binary packages built yet](https://github.com/Homebrew/brew/pull/9308)
- [`brew install ./` is now recommended for installing local file formulae](https://github.com/Homebrew/brew/pull/9241)
- [Bash, ZSH and Fish formula completion is now much faster with the `brew formulae` command](https://github.com/Homebrew/brew/pull/8846)
- [`brew install --force-bottle` refuses to ever build from source](https://github.com/Homebrew/brew/pull/9064)
- [`brew install` or `brew link` of a versioned keg-only formula will automatically unlink conflicted version formulae](https://github.com/Homebrew/brew/pull/9009)
- [`brew shellenv` is significantly faster](https://github.com/Homebrew/brew/pull/8904)
- [Homebrew has further reduced shallow clone usage (on request from GitHub)](https://github.com/Homebrew/brew/pull/8883)
- [`brew linkage` and commands using the linkage cache have significantly better performance](https://github.com/Homebrew/brew/pull/8695)
- [`brew bump-cask-pr` is a new developer command to create Homebrew/homebrew-cask pull requests](https://github.com/Homebrew/brew/pull/7986)

Finally:

> - [Discourse](https://github.com/Homebrew/brew/pull/9144) and [IRC](https://github.com/Homebrew/brew/pull/8981) are now deprecated as official communication methods in favour of [GitHub Discussions.](https://github.com/Homebrew/brew/pull/8802).

Reviewer 3: ```suggestion
- [Discourse](https://github.com/Homebrew/brew/pull/9144) and [IRC](https://github.com/Homebrew/brew/pull/8981) are now deprecated as official communication methods in favour of [GitHub Discussions](https://github.com/Homebrew/brew/pull/8802).
```

> - [Discourse](https://github.com/Homebrew/brew/pull/9144) and [IRC](https://github.com/Homebrew/brew/pull/8981) are now deprecated as official communication methods in favour of [GitHub Discussions.](https://github.com/Homebrew/brew/pull/8802).

Reviewer 6: Is this posted to discourse yet? I seem to recall the last message was "we're trying this other thing"

Author: Not posted but seems we’ve agreed on a direction. I’ll post an announcement tomorrow when I post this?
